packages/graphql/src/resolvers/IdentityEvents.js

```diff
@@ -208,17 +209,29 @@ export function identity({ id, ipfsHash }) {
           return
         }
         if (event.event === 'IdentityUpdated') {
-          ipfsHash = event.returnValues.ipfsHash
+          ipfsHashes.unshift(event.returnValues.ipfsHash)
         } else if (event.event === 'IdentityDeleted') {
-          ipfsHash = null
+          ipfsHashes = []
         }
       })
-      if (!ipfsHash) {
+      if (ipfsHashes.length < 1) {
         return resolve(null)
       }
+    } else {
+      ipfsHashes.push(ipfsHash)
     }
 
-    const data = await originIpfs.get(contracts.ipfsGateway, ipfsHash)
+    // Go through each hash until we get valid data
+    let data
+    for (const hash of ipfsHashes) {
```

Comment:
Maybe set a max of 3 to avoid too many lookups?

Reply:
sg